frontend_v2/src/app/components/challenge/challengesettings/challengesettings.component.ts

```diff
@@ -155,4 +174,237 @@ export class ChallengesettingsComponent implements OnInit {
         () => {}
       );
   }
+
+  /**
+   * Close participation function
+   */
+  stopParticipation(event) {
+    event.preventDefault();
+    const participationState = (this.challenge['is_registration_open']) ? 'Close' : 'Open';
+
+    this.apiCall = () => {
+      if (this.isChallengeHost && this.challenge['id'] !== null) {
+        const BODY = JSON.stringify({
+          'is_registration_open': !this.challenge['is_registration_open']
+        });
+        this.apiService.patchUrl(
+          this.endpointsService.editChallengeDetailsURL(this.challenge.creator.id, this.challenge.id),
+          BODY
+        ).subscribe(
+          () => {
+            this.challenge['is_registration_open'] = !this.challenge['is_registration_open'];
+            this.globalService.showToast(
+              'success', 'Participation is ' + participationState.replace('n', 'ne') + 'd successfully', 5
```

Comment:
Please create a separate string variable for this. Just looks complicated for no reason.